src/app/shared/dso-selector/dso-selector/dso-selector.component.ts

```diff
@@ -227,7 +233,8 @@ export class DSOSelectorComponent implements OnInit, OnDestroy {
         dsoTypes: this.types,
         pagination: Object.assign({}, this.defaultPagination, {
           currentPage: page
-        })
+        }),
+        sort: this.sort
```

Comment:
With a query, sorting by title is not helpful.
* I think that changing the order within the same screen can be disconcerting. Either the user is given the option to change the ordering, or it keep the same order all the time.

Sort options are configured in discovery.xml in the backend (e.g. dc.title here)
* This is a good point. I will do it as soon as it is decided how we continue.

Sorting by dc.title is not a catch-all
* Reviewing the code, I didn't see any entities other than Community/collection/item using these components. Seems safe to sort by dc.title. In any case this could be solved by sending a list of metadata to sort, instead of just one, as is done in discovery. Although it seems like a non-trivial change.

Some options:
* Apply changes only to "All Dspace" of discover, as indicated in the issue.
* Throw away the Pull Request and if someone want to change sorting must change discovery.xml, keeping the ordering homogeneous throughout the system.
* Modify the frontend so that it send ordering only if there is not a search.

Reply:
> I didn't see any entities other than Community/Collection/Item using these components

That depends on the DSpace instance you're using, there are a bunch on https://demo7.dspace.org for example
![20230126-151655](https://user-images.githubusercontent.com/31547038/214858619-51f0411f-342b-4a8d-9df1-4c0c6a782454.png)

But as @tdonohue mentioned above, we shouldn't make the order alphabetic for Items anyway.


> Modify the frontend so that it send ordering only if there is not a search

This is the way to go IMO.

> Throw away the Pull Request and if someone want to change sorting must change discovery.xml

We certainly shouldn't close this PR; it's on the right track -- sorting differently in this context is exactly what users expect.

Reply:
I'd agree here that the best solution is to modify the frontend to only send the "dc.title" ordering if there's not a search.  If the user performs a search, then the order that comes back should be based on "relevance" (that allows titles which are more relevant to appear higher in their search instead of potentially lower, if they are later alphabetically).

If we later find that users find this behavior confusing, we can always update it again in the future.  But, the simple solution for now seems to be to only order alphabetically if you haven't entered a search term. This lets you scroll down to see the alphabetic list, if you need that. Or, you can enter a search term to find the best matches based on relevance.

Reply:
Ordering has been parametriced. And it only order if you haven't entered a search term.
I need some help with the test cases requested by @ybnd. Due to my inexperience in Angular I have not been able to implement them.